lib/erb_lint/linters/hard_coded_string.rb

```diff
@@ -26,15 +26,23 @@ def run(processed_source)
         hardcoded_strings = processed_source.ast.descendants(:text).each_with_object([]) do |text_node, to_check|
           next if javascript?(processed_source, text_node)
 
-          offended_strings = text_node.to_a.select { |node| relevant_node(node) }
-          offended_strings.each do |offended_string|
-            offended_string.split("\n").each do |str|
-              to_check << [text_node, str] if check_string?(str)
+          text_node.to_a.each do |node|
+            unless string = relevant_node(node)
+              to_check << nil
```

Comment:
here we add a `nil` entry in to_check to add a clear break so next node cannot be appended to last `to_check` value